engine/table/src/main/java/io/deephaven/engine/table/impl/sort/megamerge/CharLongMegaMergeDescendingKernel.java

```diff
@@ -201,13 +205,11 @@ private static long upperBound(CharacterArraySource values, long lo, long hi, ch
     }
 
     private static long bound(CharacterArraySource valuesToSort, long lo, long hi, char searchValue,
-            @SuppressWarnings("SameParameterValue") final boolean lower) {
-        final int compareLimit = lower ? -1 : 0; // lt or leq
-
+            @SuppressWarnings("SameParameterValue") final boolean strict) {
         while (lo < hi) {
             final long mid = (lo + hi) >>> 1;
             final char testValue = valuesToSort.getUnsafe(mid);
-            final boolean moveLo = doComparison(testValue, searchValue) <= compareLimit;
+            final boolean moveLo = strict ? lt(testValue, searchValue) : leq(testValue, searchValue);
```

Comment:
I don't know if this matters a lot, but the previous version was written to branch only once, outside the while loop, instead of on every iteration. I imagine a JIT compiler can optimize this, though, and it the suppression on the method suggests that we only take one of these branches ever.